addons/stock/models/stock_move.py

```diff
@@ -1824,6 +1828,22 @@ def _create_extra_move(self):
                 extra_move = extra_move._action_confirm()
         return extra_move | self
 
+    def _check_unlink_move_dest(self):
+        """ For each move in self, check if the location_dest_id of move is outside (!= and not a child of) the source location of move_dest_ids,
+            if so, break the link.
+        """
+        new_push_moves = self.env['stock.move']
+        for move in self:
+            for move_dest in move.move_dest_ids:
+                if move.location_dest_id != move_dest.location_id and str(move_dest.location_id.id) not in move.location_dest_id.parent_path.split('/'):
+                    # break link
+                    move.move_dest_ids = [Command.unlink(move_dest.id)]
+                    move_dest.procure_method = 'make_to_stock'
+                    move_dest._recompute_state()
+                    new_push_moves |= move._push_apply()
+        if new_push_moves:
+            new_push_moves._action_confirm()
+
```

Comment:
I would do
```python
        new_push_moves = self.env['stock.move']
        for move in self:
            move_dest = move.move_dest_ids.filtered_domain(['!', ('location_id', 'child_of', move.location_dest_id.id)])
            move_dest.move_orig_ids = [Command.unlink(move.id)]
            move_dest.procure_method = 'make_to_stock'
            move_dest._recompute_state()
            new_push_moves |= move

        if new_push_moves:
            new_moves = new_push_moves._push_apply()
            new_moves._action_confirm()
```

Reply:
hmm, I am not a fan of using `filtered_domain` since I think that it could be slow, especially in this case where its not really needed.

Reply:
If you look at the `filtered_domain` function you will that with the `child_of` operator it just does a `search` :facepalm: